### ISSUE TITLE
[FEATURE] New Plagiarisms Table  to store plagiarized summary from Copydetect [Plagiarism Detector]

### ISSUE THREAD

**Issue by MohitOhlyan** [omegaup/omegaup#6688]
## New Plagiarisms Table to store plagiarized codes. 

Project Proposal:- https://docs.google.com/document/d/1H9YySiPovB7SPOOwNN1UgzUjz4ayriYqL2xmNFkCNDs/

Project Tracker:- https://docs.google.com/spreadsheets/d/1Ui-pauTRCAPsFf5J6cp9lk-NWVC6jFtvcbskG9g7Y6Y/

## Updates

We removed the idea of having `copydetect_score` and instead use only 1 field for entire content of results from copydetect. 

This is how the new datatype looks:-

{
score_A: 95%,
score_B: 92%,
green_lines: [
  [1, 5],
  [10, 20],
],
red_lines: [
 [7, 9],
 [25, 30]
],
}

**Comment by lhchavez** [comment]
can we store each of those fields as an independent column in the table? that way we can filter by runs that have a similarity score exceeding a certain threshold through a SQL query that does not involve JSON.

**Comment by MohitOhlyan** [comment]
Sure. That was the original idea. I had a meeting with Carlos and this was something that came up and sounded great. 

Will update the upcoming PR on this with those individual Fields.

**Comment by MohitOhlyan** [comment]
Although we are only storing those results which are above certain threshold.

**Comment by MohitOhlyan** [comment]
> can we store each of those fields as an independent column in the table? that way we can filter by runs that have a similarity score exceeding a certain threshold through a SQL query that does not involve JSON.

Should we keep all the similarity scores instead in the table? We are thinking of keeping only those that are above a certain threshold in the table. That way we can keep the stored usage at minimum.

**Comment by carlosabcs** [comment]
> > can we store each of those fields as an independent column in the table? that way we can filter by runs that have a similarity score exceeding a certain threshold through a SQL query that does not involve JSON.
> 
> Should we keep all the similarity scores instead in the table? We are thinking of keeping only those that are above a certain threshold in the table. That way we can keep the stored usage at minimum.

+1

**Comment by lhchavez** [comment]
> > > can we store each of those fields as an independent column in the table? that way we can filter by runs that have a similarity score exceeding a certain threshold through a SQL query that does not involve JSON.
> > 
> > 
> > Should we keep all the similarity scores instead in the table? We are thinking of keeping only those that are above a certain threshold in the table. That way we can keep the stored usage at minimum.
> 
> +1

fine with keeping all scores or only the ones that exceed a threshold, let's go with what Carlos says. something else i just realized: storing that as a separate column in the database also allows us to return the results sorted by score.

TL;DR: JSON columns are good, but we should try to limit what we store in them. storing the `green_lines` and `red_lines` in JSON columns makes perfect sense, for instance: those are only used to display stuff and never used as predicates in queries.

**Comment by MohitOhlyan** [comment]
> > > > can we store each of those fields as an independent column in the table? that way we can filter by runs that have a similarity score exceeding a certain threshold through a SQL query that does not involve JSON.
> > > 
> > > 
> > > Should we keep all the similarity scores instead in the table? We are thinking of keeping only those that are above a certain threshold in the table. That way we can keep the stored usage at minimum.
> > 
> > 
> > +1
> 
> fine with keeping all scores or only the ones that exceed a threshold, let's go with what Carlos says. something else i just realized: storing that as a separate column in the database also allows us to return the results sorted by score.
> 
> TL;DR: JSON columns are good, but we should try to limit what we store in them. storing the `green_lines` and `red_lines` in JSON columns makes perfect sense, for instance: those are only used to display stuff and never used as predicates in queries.

Understood. WIll add the new fields in the database.
